### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by throrin19**
- Add possibility to add options to fetch method like :

```javascript
myCollection.fetch({
    params : {
        name : 'Jean%',
    },
    headers : {
        Authorization : 'Bearer testbearer',
    },
});
```

- Add `toArray` method on Collection class

Linked issues : #42 #40

### CODE REVIEW

File: src/Structures/Base.js [hunk 1]
```diff
@@ -505,14 +505,17 @@ class Base {
...
      */
-    fetch() {
+    fetch(options = {}) {
         let config = () => ({
             url:     this.getFetchURL(),
             method:  this.getFetchMethod(),
-            params:  this.getFetchQuery(),
-            headers: this.getFetchHeaders(),
+            params:  _.assign({}, this.getFetchQuery(), options.params),
+            headers: _.assign({}, this.getFetchHeaders(), options.headers),
```
Comment (rtheunissen) [inline src/Structures/Base.js]: I think we could do a `defaultsDeep` here, rather than specific overrides.

File: src/Structures/Collection.js [hunk 2]
```diff
@@ -1153,6 +1153,19 @@ class Collection extends Base {
...
+     * Convert collection to Array. All models inside are converted to JSON
+     * 
+     * @return {object[]} converted collection
+     */
+    toArray() {
+        const models = [];
+
+        this.each(item => models.push(item.toJSON()));
+
+        return models;
```
Comment (rtheunissen) [inline src/Structures/Collection.js]: This could be a one-liner, I think.

`return this.map((model) => model.toJSON());`

Comment (throrin19) [reply src/Structures/Collection.js]: Yes you are right, I go change that

File: src/Structures/Base.js (hunk 1 above)
Comment (throrin19) [reply src/Structures/Base.js]: humm, like this ?

Comment (throrin19) [reply src/Structures/Base.js]: But with that, it's possible to override existing parameters if user want to do this. With `defaultsDeep`, it's not possible

Comment (rtheunissen) [reply src/Structures/Base.js]: Something like this (we actually don't need the "deep"):

```
        let config = () => _.defaults(options, {
            url:     this.getFetchURL(),
            method:  this.getFetchMethod(),
            params:  this.getFetchQuery(),
            headers: this.getFetchHeaders(),
        });
```

This allows you to override anything you want.

File: test/Structures/Collection.spec.js [hunk 3]
```diff
@@ -3204,5 +3204,19 @@ describe('Collection', () => {
...
+        it('should return an array of all toJSON models', () => {
+            let c = new Collection();
+            let m1 = c.add({id: 1});
+            let m2 = c.add({id: 2});
+
+            let cArray = c.toArray();
+
+            expect(cArray).to.be.an('array');
+            expect(cArray).to.have.lengthOf(2);
+            // expect(cArray).to.include.members([{id: 1}, {id: 2}])
```
Comment (rtheunissen) [inline test/Structures/Collection.spec.js]: We should definitely have a hard expectation here. Exactly what we expect the array to contain. We should also test that an overridden `toJSON` of a model is honoured here.

Comment (throrin19) [reply test/Structures/Collection.spec.js]: got it

File: src/Structures/Base.js (hunk 1 above)
Comment (throrin19) [reply src/Structures/Base.js]: got it

**Pull request merged**